cmd/dockerd/daemon.go

```diff
@@ -98,6 +99,17 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 	if cli.Config.Experimental {
 		logrus.Warn("Running experimental build")
 	}
+	// return human-friendly error before creating files
+	if runtime.GOOS == "linux" && os.Geteuid() != 0 {
+		return fmt.Errorf("dockerd needs to be started with root. To see how to run dockerd in rootless mode with unprivileged user see the documentation (experimental)")
```

Comment:
The same check is also done in `checksystem()`: https://github.com/moby/moby/blob/2cb26cfe9cbf8a64c5046c74d65f4528b22e67f4/daemon/daemon_unix.go#L746-L752


Which is called some lines below by `daemon.NewDaemon()` https://github.com/moby/moby/blob/2cb26cfe9cbf8a64c5046c74d65f4528b22e67f4/daemon/daemon.go#L729-L732 


- Should we update the `checksystem()` error?
- Possibly remove the check from `daemon.NewDaemon()`
- Or call the same function twice? (here, and inside daemon.NewDaemon()`)?

Reply:
removed the check from `daemon.NewDaemon`